### PR TITLE
[Code Bounty Probably Wasn't Worth It]Adds extreme options to lewd interactions

### PR DESCRIPTION
**Pull request by BobJoga**
## About The Pull Request

[Fuck](https://www.youtube.com/watch?v=0YrU9ASVw6w)

Added skullfucking options: Ear fucking and eye fucking.
Both of those account for the presence or lack of the organ.
Both of these options require both the partners to have extreme content enabled on the character setup.
Wether the recipient is damaged or not by these interactions, is defined by the extreme harm preference on the setup.

Also added footjobs because it's weird that it didn't have those but had feet grinding and licking.

## Why It's Good For The Game

it's not but at least it's optional

## Changelog
:cl:
add: Something i'm not proud of.
/:cl:

### CODE REVIEW

**Comment by CinderWC** [comment]
>Also added footjobs because it's weird that it didn't have those but had feet grinding and licking.

**Comment by ShadeAware** [comment]
An affront to God and nature.
👍

**Comment by hugmeorelse** [comment]
I am mortified yet tickled that this atrocity against humanity was coded.

On a serious note, am not sure if this is the sort of image we want to go for as a server. Personally, I think extreme content should be unviewable unless a toggle is made in preferences, because I don't think a lot of onlookers want to witness something like this at all, particularly if they're ghosting.

**Comment by BobJoga** [comment]
> I am mortified yet tickled that this atrocity against humanity was coded by someone willing to risk their soul to make a bounty.
> 
> _On a serious note, I am not sure if this is the sort of image we want to go for as a server._ On the flip side, I don't see a problem with providing more options, provided we do this right. Freedom is good after all, but I don't know.. I think having it be toggleable is an important first step, but we need to do more.
> 
> **Personally, I think extreme content shouldn't be viewable by others unless a toggle is made in preferences.** I am not a betting man, but I am willing to gamble that people do not want to see this lighting up their chat when ghosting.
> 
> **So yeah, I don't think I'll be merging this without at least discussing that safeguard for folks.** Eyeballs must be preserved, people. Even then, I'm not so sure.
> 
> **Note:** I am going to draw some lines though. No bathroom kinks among other things.

I'll probably try and do something to hide this atrocity from unwilling players. I hadn't thought about it but yeah, i myself wouldn't like to see this popping up in my chat as a ghost.

**Comment by hugmeorelse** [comment]
> > I am mortified yet tickled that this atrocity against humanity was coded by someone willing to risk their soul to make a bounty.
> > _On a serious note, I am not sure if this is the sort of image we want to go for as a server._ On the flip side, I don't see a problem with providing more options, provided we do this right. Freedom is good after all, but I don't know.. I think having it be toggleable is an important first step, but we need to do more.
> > **Personally, I think extreme content shouldn't be viewable by others unless a toggle is made in preferences.** I am not a betting man, but I am willing to gamble that people do not want to see this lighting up their chat when ghosting.
> > **So yeah, I don't think I'll be merging this without at least discussing that safeguard for folks.** Eyeballs must be preserved, people. Even then, I'm not so sure.
> > **Note:** I am going to draw some lines though. No bathroom kinks among other things.
> 
> I'll probably try and do something to hide this atrocity from unwilling players. I hadn't thought about it but yeah, i myself wouldn't like to see this popping up in my chat as a ghost.

Yeah, if I'm going to be honest, I may require coding bounties to be approved by myself and the maintainers for quality control and to save prospective bounty hunters time in the future. I think it's good to let individual players contribute parts of what they want to see in the server, but at some point, it starts to conflict with the direction we want to take the server. I'm only considering this simply because you put the work into it, and we haven't really made a policy on how coding bounties will be managed.

**Comment by ForrestWick** [comment]
> > > I am mortified yet tickled that this atrocity against humanity was coded by someone willing to risk their soul to make a bounty.
> > > _On a serious note, I am not sure if this is the sort of image we want to go for as a server._ On the flip side, I don't see a problem with providing more options, provided we do this right. Freedom is good after all, but I don't know.. I think having it be toggleable is an important first step, but we need to do more.
> > > **Personally, I think extreme content shouldn't be viewable by others unless a toggle is made in preferences.** I am not a betting man, but I am willing to gamble that people do not want to see this lighting up their chat when ghosting.
> > > **So yeah, I don't think I'll be merging this without at least discussing that safeguard for folks.** Eyeballs must be preserved, people. Even then, I'm not so sure.
> > > **Note:** I am going to draw some lines though. No bathroom kinks among other things.
> > 
> > 
> > I'll probably try and do something to hide this atrocity from unwilling players. I hadn't thought about it but yeah, i myself wouldn't like to see this popping up in my chat as a ghost.
> 
> Yeah, if I'm going to be honest, **Looking at this now, I may require coding bounties to be approved by myself and the maintainers for quality control and to save prospective bounty hunters time in the future.** I think it's good to let individual players contribute parts of what they want to see in the server, but at some point, it starts to conflict with the direction we want to take the server. I don't want time and money to be wasted.
> 
> I'm only considering this simply because we haven't formed an actual policy on bounties yet, especially since you put some work into it and had the sense to come up with some safeguards for players. Even then, I'm not sure if I want this merged at the end of the day. I may have to talk with others about this. We'll see. I might just be overreacting.

The terms of a bounty, traditionally, are you code it, you get paid. Merging doesn't matter. The bounty poster takes and acknowledges that risk in the end. Also, I don't see how feet grinding is fine but then earfucking is going too far. If Bob wants, he can take out eye fucking though since that was a meme.

**Comment by Strazyplus** [comment]
> oh god 
> > why?
 > > >   i myself wouldn't like to see this popping up in my chat as a ghost.
### **Please.**

**Comment by Andromeda-K22** [comment]
I'm pretty certain this is actually against BYOND's terms of service, bravo.

**Comment by BobJoga** [comment]
> I'm pretty certain this is actually against BYOND's terms of service, bravo.

Honestly no idea and this is probably not going to be merged.

Besides, lifeweb exists.

**Comment by BobJoga** [comment]
> I'm pretty certain this is actually against BYOND's terms of service, bravo.

just checked the byond ToS
ERP servers technically shouldn't even exist lol
image
whether or not this PR is merged really doesn't matter, either way we can technically be shut down

**Comment by BobJoga** [comment]
> > > I am mortified yet tickled that this atrocity against humanity was coded by someone willing to risk their soul to make a bounty.
> > > _On a serious note, I am not sure if this is the sort of image we want to go for as a server._ On the flip side, I don't see a problem with providing more options, provided we do this right. Freedom is good after all, but I don't know.. I think having it be toggleable is an important first step, but we need to do more.
> > > **Personally, I think extreme content shouldn't be viewable by others unless a toggle is made in preferences.** I am not a betting man, but I am willing to gamble that people do not want to see this lighting up their chat when ghosting.
> > > **So yeah, I don't think I'll be merging this without at least discussing that safeguard for folks.** Eyeballs must be preserved, people. Even then, I'm not so sure.
> > > **Note:** I am going to draw some lines though. No bathroom kinks among other things.
> > 
> > 
> > I'll probably try and do something to hide this atrocity from unwilling players. I hadn't thought about it but yeah, i myself wouldn't like to see this popping up in my chat as a ghost.
> 
> Yeah, if I'm going to be honest, **Looking at this now, I may require coding bounties to be approved by myself and the maintainers for quality control and to save prospective bounty hunters time in the future.** I think it's good to let individual players contribute parts of what they want to see in the server, but at some point, it starts to conflict with the direction we want to take the server. I don't want time and money to be wasted.
> 
> I'm only considering this simply because we haven't formed an actual policy on bounties yet, especially since you put some work into it and had the sense to come up with some safeguards for players. Even then, I'm not sure if I want this merged at the end of the day. I may have to talk with others about this. We'll see. I might just be overreacting.

I have updated the PR now. Extreme content will not appear in the chat at all for people who have it set to "No". I have also done some general improvements on interaction code, and people who have lewd verbs disabled won't see lewd interactions happen (none of them).

**Comment by Leshana** [comment]
@BobJoga 
image

**Comment by CinderWC** [comment]
> 
> 
> @BobJoga
> image

Look at the Vorestation Contributor complain about someone adding more ERP verb content lmao.
Imagine lacking this much self-awareness

**Comment by Leshana** [comment]
> 
> Look at the Vorestation Contributor complain about someone adding more ERP verb content lmao.
> Imagine lacking this much self-awareness

Imagine me posting an image specifically for the purpose of showing ERP content is **NOT** a BYOND TOS violation.  A post without a hint of complaint about the PR content!

**Comment by BobJoga** [comment]
> @BobJoga
> image

no i dont know how to read

**Comment by Strazyplus** [comment]
H O N K

**Comment by hugmeorelse** [comment]
This is getting too toxic. Either discuss like adults or leave the conversation.

**Comment by Commandersand** [comment]
Glad that Vore and Erotic Roleplay isn't pornographic or vulgar

**Comment by BobJoga** [comment]
why the fuck do i even bother with this terrible PR

**Comment by Commandersand** [comment]
because you're funny
this pr is funny

**Comment by hugmeorelse** [comment]
> because you're funny
> this pr is funny

Bob is the King of Funny.

**Comment by Leshana** [comment]
> 
> > I'm pretty certain this is actually against BYOND's terms of service, bravo.
> 
> just checked the byond ToS
> ERP servers technically shouldn't even exist lol
> [image]
> whether or not this PR is merged really doesn't matter, either way we can technically be shut down
I believe you have nothing to worry about.  BYOND's TOS doesn't mention anything about game content, only discussing content posted on the byond website, forums, etc, even in the section you have highlighted.

**Comment by Afya212** [comment]
This chat, oh my gosh.

**Comment by Commandersand** [comment]
do people even use the byond forums/website :think:

**Comment by BobJoga** [comment]
Ok now being completely serious here, this PR started as a pretty simple code bounty but by now is just pretty much evolving to an almost complete overhaul to the interaction menu - i should probably close this and create another PR that reflects the nature better other than just highlighting the part i find myself to be a bit untastesful.

**Comment by hugmeorelse** [comment]
> Ok now being completely serious here, this PR started as a pretty simple code bounty but by now is just pretty much evolving to an almost complete overhaul to the interaction menu - i should probably close this and create another PR that reflects the nature better other than just highlighting the part i find myself to be a bit untastesful.

Probably a good idea.

**Comment by CinderWC** [comment]
@Leshana Your only contribution to this entire repo is coming in here to ruleslawyer and complain about ERP content on our server when you, yourself work for Vorestation, which is literally all about cringy ERP lmao.
Please work on your self awareness i cant take you any seriously bruh

File: modular_skyrat/code/interactions/lewd/_lewd.dm [hunk 1]
```diff
@@ -13,6 +18,11 @@
 #define NORMAL_LUST 10
 #define LOW_LUST 1
 
+#define REQUIRE_NONE 0
+#define REQUIRE_EXPOSED 1
+#define REQUIRE_ANY 2
+#define REQUIRE_UNEXPOSED 3
```
Comment (Leshana) [inline modular_skyrat/code/interactions/lewd/_lewd.dm:24]: If you change these values as such, you lose nothing but gain the option for bit math comparisions in the future if becomes convenient.
```suggestion
#define REQUIRE_NONE 0
#define REQUIRE_EXPOSED 1
#define REQUIRE_UNEXPOSED 2
#define REQUIRE_ANY 3
```

**Comment by hugmeorelse** [comment]
I'm closing the comment section, because people are choosing to beat a dead horse over this and be rude.

@Leshana Bob will be opening a more appropriate PR if you're interested in reviewing his code there.

**Pull request closed without merging**